assets/js/modules/analytics-4/components/settings/AdsConversionIDSettingsNotice.js

```diff
@@ -45,6 +47,35 @@ export default function AdsConversionIDSettingsNotice() {
 	const adsConversionIDMigratedAtMs = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS_4 ).getAdsConversionIDMigratedAtMs()
 	);
+	const viewContext = useViewContext();
+	const trackDismissNotificationEvent = () => {
+		trackEvent(
+			`${ viewContext }_GA_Ads_redirect`,
+			'dismiss_notification'
+		);
+	};
+	const trackConfirmNotificationEvent = () => {
+		trackEvent(
+			`${ viewContext }_GA_Ads_redirect`,
+			'confirm_notification'
+		);
+	};
+
+	// Track a view_notification event.
+	useMount( () => {
+		// Only track the view event if the notice display condition is satisfied.
+		// The valid condition is a data migration within the 28 day window.
+		if (
+			adsConversionIDMigratedAtMs &&
+			Date.now() - adsConversionIDMigratedAtMs <
+				28 * DAY_IN_SECONDS * 1000
+		) {
+			trackEvent(
+				`${ viewContext }_GA_Ads_redirect`,
+				'view_notification'
+			);
+		}
+	} );
```

Comment:
This looks much better now. However, I think we missed one additional scenario where the `trackEvent` call could still be fired without the notice appearing, and that is when the notice is dismissed.

I think we should prevent `trackEvent` here from being called if the `ADS_CONVERSION_ID_NOTICE_DISMISSED_ITEM_KEY` item is dismissed, i.e. using the `core/user` `isItemDismissed` selector.

Reply:
@nfmohit this has now been addressed. There was a lot of back and forth between @techanvil and I, and @techanvil eventually implemented a wrapper for notices that require inView callbacks. This was mostly due to the fact that the useSelect would not be updated with a ref item was updated, which it was when the inner SettingsNotice component would render, and passing the ref.current object as a dep to useSelect is an anti-pattern. @techanvil has a lot more info on this if we need to expand, but we've arrived at a working solution here.

Reply:
Thanks @10upsimon. Here's the related [Slack conversation](https://10up.slack.com/archives/CBKKQEBR9/p1712922609360979?thread_ts=1712732138.204779&cid=CBKKQEBR9) for additional context.

A couple of points:
- Simon, if you haven't already, please can you file a followup issue to provide an improved version of `useIntersection()` and replace our current use of the version from `react-use`?
- Additionally, it might be worth addressing this comment now, rather than leaving this tucked away in `SettingsNotice`. Mind you if we have the followup issue then maybe it's not worth it as `SettingsNoticeWithIntersectionObserver` can be removed as part of that issue.
https://github.com/google/site-kit-wp/blob/b8d25c16a5e65fb15a8bbf34a88b2273e5d578ed/assets/js/components/SettingsNotice/SettingsNoticeWithIntersectionObserver.js#L35-L37

Reply:
I agree. Let's address the TODO comment here. I've raised it [here](https://github.com/google/site-kit-wp/pull/8474#discussion_r1566471289). Thanks!

Reply:
This has been addressed in https://github.com/google/site-kit-wp/pull/8474/commits/32a7df0382754fe09410321799bcb4cb0da95986 and commits that followed it.